scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -69,12 +95,24 @@
     (assoc expected-state-dict :launch-pod [:elided-for-brevity])
     expected-state-dict))
 
-;; REVIEW: Review handle-status-update
 (defn pod-has-started
   "A pod has started."
-  [{:keys [synthesized-state pod] :as existing-state-dictionary}]
-  ;(api/TODO); TODO Update datomic state to instance.state/running.
-  {:expected-state :expected/running})
+  [kcc {:keys [pod] :as existing-state-dictionary}]
+  (let [task-id (-> pod .getMetadata .getName)
+        status {:task-id {:value task-id}
+                :state :task-running}]
+    (handle-status-update kcc status)
+    {:expected-state :expected/running}))
+
+(defn pod-was-killed
+  [kcc {:keys [pod] :as existing-state-dictionary}]
+  (let [task-id (-> pod .getMetadata .getName)
+        status {:task-id {:value task-id}
+                :state :task-failed
+                :reason :reason-command-executor-failed}]
```

Comment:
Is this the right reason?

Reply:
Currently, we just get these back via the normal failed job flow once the command gets killed. We could do something more specific, but I think it's better to be consistent.

Reply:
I don't like, but OK.